presidio-anonymizer/tests/test_anonymizer_engine.py

```diff
@@ -255,3 +256,46 @@ def _operate(
     return EngineResult(
         "Number: I am your new text!", [OperatorResult(0, 35, "type", "text", "hash")]
     )
+
+
+def test_when_text_entities_conflict_selected_then_all_conflicts_handled():
+    engine = AnonymizerEngine()
+    text = (
+        "Fake card number 4151 3217 6243 3448.com that "
+        "overlaps with nonexisting URL."
+    )
+    analyzer_result = RecognizerResult("CREDIT_CARD", 17, 36, 0.8)
+    analyzer_result2 = RecognizerResult("URL", 32, 40, 0.8)
+    anonymizer_config = OperatorConfig("keep")
+    conflict_strategy = ConflictResolutionStrategy.TEXT_AND_ENTITIES
+    result = engine.anonymize(
+        text,
+        [analyzer_result, analyzer_result2],
+        {"DEFAULT": anonymizer_config},
+        conflict_resolution=conflict_strategy
+    ).text
+
+    assert result == text
+
+
+def test_when_none_conflict_selected_then_no_conflict_handled():
```

Comment:
Can you please add another case where conflict_stragy=None (which is the default)?

If the tests are identical, but results are different, consider using pytest.mark.parameterize

Reply:
Hi @omri374 
Added one more test case where conflict_strategy=None. Please let me know if any further changes.